addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3678,6 +3688,154 @@ options.registry.WebsiteAnimate = options.Class.extend({
     },
 });
 
+/**
+ * Allows edition of text "Highlight Effects" following this generic structure:
+ * `<span class="o_text_highlight">
+ *      <span class="o_text_highlight_item">
+ *          line1-textNode1 [line1-textNode2,...]
+ *          <svg.../>
+ *      </span>
+ *      [<br/>]
+ *      <span class="o_text_highlight_item">
+ *          line2-textNode1 [line2-textNode2,...]
+ *          <svg.../>
+ *      </span>
+ *      ...
+ * </span>`
+ * To correctly adapt each highlight unit when the text content is changed.
+ */
+options.registry.TextHighlight = options.Class.extend({
```

Comment:
If if save a snippet with text highlight in it (with the `snippetSave` option), when I drop this saved snippet onto the page, the text is no longer highlighted until a resize.